tests/llm/test_request_builder.py

```
@@ -95,3 +97,48 @@ async def test_provider_params_are_merged_into_extra_params(
+    timeout: object,
+) -> None:
+    """Invalid provider timeout values fail before provider SDK calls."""
+    config = ModelConfig(
+        model="gpt-4.1-mini",
+        transport="openai",
+        provider_params={"timeout": timeout},
+    )
+
+    with pytest.raises(
+        ValidationException,
+        match="provider_params.timeout must be a positive number of seconds",
```

Ruff (0.15.18)

[warning] Pattern passed to `match=` contains metacharacters but is neither escaped nor raw

(RUF043)